massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -860,7 +860,8 @@ impl ProtocolWorker {
     /// Prune `checked_headers` if it is too large
     fn prune_checked_headers(&mut self) {
         if self.checked_headers.len() > self.config.max_known_blocks_size {
-            self.checked_headers.clear();
+            self.checked_headers
+                .drain_filter(|id, _| !self.block_wishlist.contains_key(id));
```

Comment:
I don't understand, I think this basically prevents cache cleanup but doesn't actually retain the object